src/main/groovy/org/boozallen/plugins/jte/init/primitives/TemplatePrimitive.groovy

```diff
@@ -16,57 +16,63 @@
 package org.boozallen.plugins.jte.init.primitives
 
 import com.cloudbees.groovy.cps.NonCPS
+import org.boozallen.plugins.jte.util.JTEException
+import org.boozallen.plugins.jte.util.TemplateLogger
+import org.jenkinsci.plugins.workflow.cps.CpsScript
+import org.jenkinsci.plugins.workflow.cps.GlobalVariable
+
+import javax.annotation.Nonnull
 
 /**
- * Objects whose class extends TemplatePrimitive will be protected in the {@link TemplateBinding} from
- * being inadvertently overridden
+ * Framework constructs that make templates easier to write.
+ * Typically created by parsing the Pipeline Configuration.
  */
-abstract class TemplatePrimitive implements Serializable{
+abstract class TemplatePrimitive extends GlobalVariable implements Serializable{
 
     private static final long serialVersionUID = 1L
```

Comment:
The `TemplatePrimitive` only gets persisted to a job as part of the binding.  This wouldn't be unmarshalled because if someone were to replay the job we'd create a new binding anyway.